lib/config/authmode.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package config
+
+type AuthMode int
+
+const (
+	AuthModeStatic AuthMode = iota // default is static
+	AuthModeLDAP
```

Comment:
I would add a None mode and stop checking for username and password from being set, and do a migration in the config setting auth mode.

This would also need fixes in the UI.

Reply:
@calmh wdyt?

Reply:
That would seem clean but if that were the default it would mean existing things that just set username and password would result in no auth which is bad.